core/embed/trezorhal/stm32u5/haptic/drv2625/drv2625.c

```diff
@@ -136,19 +140,25 @@ void haptic_init(void) {
   TIM16->BDTR |= TIM_BDTR_MOE;
 }
 
-static void haptic_play_RTP(int8_t amplitude, uint16_t duration_ms) {
-  set_reg(DRV2625_REG_MODE,
-          DRV2625_REG_MODE_RTP | DRV2625_REG_MODE_TRGFUNC_ENABLE);
-  set_reg(DRV2625_REG_RTP, (uint8_t)amplitude);
+static bool haptic_play_RTP(int8_t amplitude, uint16_t duration_ms) {
+  bool success = true;
+  success &= set_reg(DRV2625_REG_MODE,
```

Comment:
As `set_reg()` returns a bool, we should avoid using the `&` operator on it (nor `&=`). I understand that it may work, but it's not considered good practice. Furthermore, if the first of `set_reg()` call fails, the routine continues and calls set_reg() again and it finally starts the timer...

In this case, I would recommend:

```cpp
if (!set_reg(...)) {
    return false;
}

if (!set_reg(...)) {
    return false;
}
```

or

```cpp
success = set_reg(...);
success = success && set_reg(...);
if (!success) {
    return false;
}
```

Reply:
fixed https://github.com/trezor/trezor-firmware/pull/3495/commits/92abdb587ff27255e27fac598b4b7ddc3dad16d0